zkvm/src/vm.rs

```diff
@@ -281,6 +281,33 @@ where
         Ok(())
     }
 
+    fn r#const(&mut self) -> Result<(), VMError> {
+        let data = self.pop_item()?.to_data()?.to_bytes();
+        let mut slice = Subslice::new(&data);
+        let scalar = slice.read_scalar()?;
+        self.push_item(Expression::constant(scalar));
+        Ok(())
+    }
+
+    fn var(&mut self) -> Result<(), VMError> {
+        let comm = self.pop_item()?.to_data()?.to_commitment()?;
+        let v = self.make_variable(comm);
+        self.push_item(v);
+        Ok(())
+    }
+
+    fn mintime(&mut self) -> Result<(), VMError> {
+        let a: Scalar = self.mintime.into();
+        self.push_item(Expression::constant(a));
```

Comment:
If `Expression::constant` is generic over `S: Into<Scalar>`:
```suggestion
        self.push_item(Expression::constant(self.mintime));
```